app/backend/approaches/chatreadretrieveread.py

```diff
@@ -194,6 +194,8 @@ async def run(self, history: Sequence[dict[str, str]], overrides: dict[str, Any]
                     # max_tokens=32, # setting it too low may cause malformed JSON
                     max_tokens=100,
                 n=1)
+            if not chat_completion or not chat_completion.choices[0].message.content:
+                raise ValueError("No choices returned in chat completion response")
```

Comment:
Put the same check on the second call to Azure OpenAI (line 386)

Reply:
updated all to raise the filtered message if hit.